pkg/riff/commands/function.go

```diff
@@ -29,7 +29,23 @@ func NewFunctionCommand(ctx context.Context, c *cli.Config) *cobra.Command {
 		Use:   "function",
 		Short: "functions built from source using function buildpacks",
 		Long: strings.TrimSpace(`
-<todo>
+Functions are a mechanism for converting language idiomatic units of logic into
+container images that can be HTTP invoked or process streams of messages. Cloud
+Native Buildpacks are provided to detect the language, provide a language
+runtime, install build and runtime dependencies, compile the function, and
+package it into a container.
+
+The function resource is only responsible for converting source code into a
+container. The function container image may then be deployed as a request
+handler, or as a stream processor.
+
+Functions are distinct from applications in the scope of the source code. Unlike
```

Comment:
Not sure what "in the scope of the source code" is getting at.

Reply:
The application and function resource reference source code and define how to build the source.